src/Api/Serializer/RestSerializer.php

```diff
@@ -126,6 +126,15 @@ private function applyHeader($name, Shape $member, $value, array &$opts)
         if ($member->getType() == 'timestamp') {
             $value = TimestampShape::format($value, 'rfc822');
         }
+        if ($member['jsonvalue']) {
+            $value = json_encode($value);
+            if ($value === null && JSON_ERROR_NONE !== json_last_error()) {
+                throw new \ParserException('Unable to encode the provided value'
```

Comment:
Have updated the exception type to: InvalidArgumentException.

This is only added as a safety net as the error will only be thrown if the value is a resource.